src/robotFactory.js

```diff
@@ -1,10 +1,70 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
```suggestion
  constructor(
    name, 
    weight, 
    { x = 0, y = 0 }, 
    chipVersion,
  ) {
```
Better use a modern approach of setting default values using object restructuring.
Also, separate arguments with new lines if you have more than 3 ones.

Reply:
done